pkg/sdk/sql_builder.go

```diff
@@ -7,82 +7,149 @@ import (
 	"unsafe"
 )
 
-// couple of helper functions.
-func parentheses(s string) string {
-	return fmt.Sprintf("(%s)", s)
+type modifierType string
+
+const (
+	quoteModifierType   modifierType = "quotes"
+	parenModifierType   modifierType = "paren"
+	reverseModifierType modifierType = "reverse"
+	equalsModifierType  modifierType = "equals"
+)
+
+type modifier interface {
+	Modify(v any) string
 }
 
-type quoteType string
+type quoteModifier string
 
 const (
-	NoQuotes     quoteType = "no_quotes"
-	DoubleQuotes quoteType = "double_quotes"
-	SingleQuotes quoteType = "single_quotes"
+	NoQuotes     quoteModifier = "no_quotes"
+	DoubleQuotes quoteModifier = "double_quotes"
+	SingleQuotes quoteModifier = "single_quotes"
 )
 
-func (qt quoteType) Quote(v interface{}) string {
+func (qm quoteModifier) Modify(v any) string {
 	s := fmt.Sprintf("%v", v)
-	switch qt {
+	switch qm {
 	case NoQuotes:
 		return s
 	case DoubleQuotes:
-		escapedString := strings.ReplaceAll(s, qt.String(), qt.String()+qt.String())
-		return fmt.Sprintf(`%v%v%v`, qt.String(), escapedString, qt.String())
+		escapedString := strings.ReplaceAll(s, qm.String(), qm.String()+qm.String())
+		return fmt.Sprintf(`%v%v%v`, qm.String(), escapedString, qm.String())
 	case SingleQuotes:
-		escapedString := strings.Trim(s, qt.String())
-		return fmt.Sprintf(`%v%v%v`, qt.String(), escapedString, qt.String())
+		// https://docs.snowflake.com/en/sql-reference/data-types-text#single-quoted-string-constants
+		// replace all single quotes with \'
+		escapedString := strings.ReplaceAll(s, qm.String(), `\'`)
```

Comment:
Single quotes use the same escaping strategy as double quotes, they're doubled.
```suggestion
		escapedString := strings.ReplaceAll(s, qm.String(), qm.String()+qm.String())
```